Initial commit to new React UI

Nguyen-The-Lap opened the pull request:
<img width="1910" height="928" alt="image" src="https://github.com/user-attachments/assets/a17aa6c3-a930-496e-9fd0-759fd5c5b285" />

Nguyen-The-Lap committed:
Initial commit to new React UI

Nguyen-The-Lap committed:
FIxing clogging elements

Nguyen-The-Lap commented:
<img width="1902" height="929" alt="image" src="https://github.com/user-attachments/assets/160bc12c-5ea8-4dbd-8e12-9f4cf9fcc05b" />
<img width="1887" height="923" alt="image" src="https://github.com/user-attachments/assets/47ba2081-8dd0-4810-bdea-a3af6f0b93e7" />
<img width="1899" height="921" alt="image" src="https://github.com/user-attachments/assets/888385df-23e1-49e0-b92f-50acc44c4f50" />
<img width="1891" height="923" alt="image" src="https://github.com/user-attachments/assets/7e7fb580-2de2-4669-ae5a-8baf333bf4a6" />
<img width="1899" height="928" alt="image" src="https://github.com/user-attachments/assets/24aa5ca8-4275-40b1-a78c-1d76628f8967" />

Nguyen-The-Lap commented:
<img width="1906" height="496" alt="image" src="https://github.com/user-attachments/assets/27c683ff-567b-403e-ae18-71871db24244" />

Nguyen-The-Lap committed:
Modify InfoCards title and description text

Updated card title and description for InfoCards component.

nquynh06 commented:
<img width="1910" height="928" alt="Group 1047" src="https://github.com/user-attachments/assets/c2b629fd-7f26-4a51-94c9-c8225daf8c1c" />
Tui thấy tên nhóm nên để to hơn xíu với lại navbar góc phải trên các button nên cho size chữ nhỏ lại xíu(tham khảo ảnh này tui mới sửa lại nha)

nquynh06 commented:
với lại logo góc trái trên cùng bị lệch với vòng tròn nè, nếu nó ko vào giữa đc thì bỏ vòng tròn luôn 
<img width="585" height="490" alt="image" src="https://github.com/user-attachments/assets/67194e31-ecbf-49cb-822c-8d8bd2a830fb" />

nquynh06 commented:
với lại theme chính là màu tím roi thì nên cho scroll bar màu tím luôn nha @Nguyen-The-Lap

Nguyen-The-Lap committed:
fix small problems

Nguyen-The-Lap commented:
<img width="1916" height="941" alt="image" src="https://github.com/user-attachments/assets/5c2a6c67-6980-4ef2-a3a9-5e387353decb" />

nquynh06 commented:
<img width="853" height="166" alt="image" src="https://github.com/user-attachments/assets/d8f99c9a-5edd-4766-bae3-0bf7a62cec20" />
cho size chữ chỗ này nhỏ lại như cái hình tui review phía trên á

nquynh06 commented:
Tui thấy tiêu đề mấy mục lớn phía dưới ông dùng tiếng anh thì cái "giới thiệu" này cũng nên dùng tiếng anh
<img width="154" height="60" alt="image" src="https://github.com/user-attachments/assets/5307c742-4e28-4fec-a37f-f3ab58f8822f" />

Nguyen-The-Lap commented:
<img width="1915" height="953" alt="image" src="https://github.com/user-attachments/assets/31beee1f-a585-4ed8-b4e8-e22fa8f33972" />
<img width="1425" height="855" alt="image" src="https://github.com/user-attachments/assets/96ad960e-69b1-468c-97e4-6c39ff3a14eb" />

Nguyen-The-Lap committed:
feat: add automatic year update in footer and performance optimizations

- Automatically update copyright year using JavaScript's Date object
- Add service worker for offline support and asset caching
- Implement code splitting and lazy loading for better performance
- Add resource hints and preconnect for faster loading
- Update performance configuration with feature flags
- Add workbox configuration for service worker generation
- Fix linting warnings and improve code organization

nquynh06 commented:
<img width="859" height="153" alt="image" src="https://github.com/user-attachments/assets/a2888471-1a65-46c0-a028-2762e76f29a8" />
cho chỗ navbar text nhỏ hơn nữa nha(nhỏ như hình) tại vẫn còn hơi to nhìn nó vẫn sát nhau @Nguyen-The-Lap

Nguyen-The-Lap committed:
style: improve responsive layout for mobile devices

- Make info cards stack vertically on small screens
- Optimize navigation bar for mobile with horizontal scrolling
- Reduce padding and spacing for better mobile experience
- Adjust font sizes and touch targets for mobile usability
- Maintain desktop layout while enhancing mobile responsiveness

Nguyen-The-Lap commented:
style: improve responsive layout for mobile devices

- Make info cards stack vertically on small screens
- Optimize navigation bar for mobile with horizontal scrolling
- Reduce padding and spacing for better mobile experience
- Adjust font sizes and touch targets for mobile usability
- Maintain desktop layout while enhancing mobile responsiveness



<img width="1320" height="874" alt="image" src="https://github.com/user-attachments/assets/31f8fd44-8ea9-4901-a87f-d965e874dfad" />
<img width="1173" height="414" alt="image" src="https://github.com/user-attachments/assets/43afcc98-2ac5-4a20-85fa-b106f55169c1" />
<img width="1917" height="937" alt="image" src="https://github.com/user-attachments/assets/02b28624-d3ec-44ab-bc2f-cad45049919a" />
